#! /usr/bin/env python

PACKAGE='image_cb_detector'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                  Type      Level   Description                   Default   Min   Max
gen.add("num_x",              int_t,    0,      "Number of columns",          8,        1,    30)
gen.add("num_y",              int_t,    0,      "Number of rows",             6,        1,    30)
gen.add("dim_x",              double_t, 0,      "Space between columns",      1,        0,    1)
gen.add("dim_y",              double_t, 0,      "Space between rows",         1,        0,    1)
gen.add("adaptive_threshold", bool_t,   0,      "Use adaptive thresholding to convert the image to black and white rather than a fixed threshold level",  False)
gen.add("filter_quads",       bool_t,   0,      "Use additional criteria to filter out false quads extracted at the contour retrieval stage", False)
gen.add("normalize",          bool_t,   0,      "Normalize the image gamma with equalizeHist() before applying fixed or adaptive thresholding", False)
gen.add("fast",               bool_t,   0,      "Run a fast check on the image that looks for chessboard corners, and shortcut the call if none is found", False)

exit(gen.generate(PACKAGE, "image_cb_detector", "Chessboard"))
